verbs: Add support for fi_trywait call.

a-ilango opened the pull request.

shefty commented:
Can you please re-push this PR, so that it can get a fresh travis run?

a-ilango commented:
Done.

shefty commented on prov/verbs/src/verbs_domain.c:
```diff
@@ -256,6 +256,36 @@ fi_ibv_domain(struct fid_fabric *fabric, struct fi_info *info,
+
+		ret = trywait(fids[i]);
+		if (ret)
+			return ret;
```
I would drop the trywait variable and just call cq->trywait directly in the switch statement.

a-ilango replied:
Done.

shefty commented on prov/verbs/src/verbs_cq.c:
```diff
@@ -345,6 +345,53 @@ int fi_ibv_cq_signal(struct fid_cq *cq)
+		goto err;
+	}
+
+	rc = ibv_req_notify_cq(cq->cq, 0);
```
Where are the CQ events read from the kernel and acked?

a-ilango commented:
Added call to read CQ events in trywait and to improve performance acknowledgement of CQ events is delayed until the CQ is closed. Refactored cq_sread to make use of fi_trywait

shefty commented:
:+1:
